src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js

```diff
@@ -273,6 +306,340 @@ function hasArguments() {
+}
+
+function makeAvailabilityPath(clusterName, options) {
+  if (options.hash.struct) {
+    if (options.hash.structField) {
+      return [
+        'struct fields',
+        clusterName,
+        options.hash.struct,
+        options.hash.structField,
+      ];
+    }
+
+    return ['structs', clusterName, options.hash.struct];
+  }
+
+  if (options.hash.event) {
+    if (options.hash.eventField) {
+      return [
+        'event fields',
+        clusterName,
+        options.hash.event,
+        options.hash.eventField,
+      ];
+    }
+
+    return ['events', clusterName, options.hash.event];
+  }
+
+  if (options.hash.command) {
+    if (options.hash.commandField) {
+      return [
+        'command fields',
+        clusterName,
+        options.hash.command,
+        options.hash.commandField,
+      ];
+    }
+
+    return ['commands', clusterName, options.hash.command];
+  }
+
+  if (options.hash.attribute) {
+    return ['attributes', clusterName, options.hash.attribute];
+  }
+
+  if (options.hash.enum) {
+    if (options.hash.enumValue) {
+      return [
+        'enum values',
+        clusterName,
+        options.hash.enum,
+        options.hash.enumValue,
+      ];
+    }
+
+    return ['enums', clusterName, options.hash.enum];
+  }
+
+  if (options.hash.bitmap) {
+    if (options.hash.bitmapValue) {
+      return [
+        'bitmap values',
+        clusterName,
+        options.hash.bitmap,
+        options.hash.bitmapValue,
+      ];
+    }
+
+    return ['bitmaps', clusterName, options.hash.bitmap];
+  }
+
+  if (options.hash.api) {
+    return ['apis', options.hash.api];
+  }
+
+  if (options.hash.globalAttribute) {
+    return ['global attributes', options.hash.globalAttribute];
+  }
+
+  return ['clusters', clusterName];
+}
+
+async function availability(clusterName, options) {
+  const data = fetchAvailabilityData(this.global);
+  const path = makeAvailabilityPath(clusterName, options);
+
+  if (
+    options.hash.fabricScopedDeprecationMessage &&
+    options.hash.nonFabricScopedDeprecationMessage &&
+    options.hash.type
+  ) {
+    // Figure out the right deprecation message for cases where it unfortunately
+    // depends on the type.
+    if (options.hash.deprecationMessage) {
+      throw new Error(
+        `Should not specify deprecationMessage along with fabricScopedDeprecationMessage and nonFabricScopedDeprecationMessage`
+      );
+    }
+    let packageIds = await templateUtil.ensureZclPackageIds(this);
+    let st = await zclQuery.selectStructByName(
+      this.global.db,
+      options.hash.type,
+      packageIds
+    );
+    if (st && st.isFabricScoped) {
+      options.hash.deprecationMessage =
+        options.hash.fabricScopedDeprecationMessage;
+    } else {
+      options.hash.deprecationMessage =
+        options.hash.nonFabricScopedDeprecationMessage;
+    }
+  }
+
+  // This assumes that an item is introduced in some single release (which
```

Comment:
Is there a straightforward way to catch if this is violated and throw an error?

Reply:
`findReleaseForPath` could keep looking instead of returning once if finds something, and throw if it finds a second thing.

Reply:
`findReleaseForPath` could keep looking instead of returning once if finds something, and throw if it finds a second thing.

Reply:
Made that change.